src/svd/interrupt.rs

```diff
@@ -0,0 +1,50 @@
+use crate::chip;
+use crate::DisplayError;
+use crate::elementext::ElementExt;
+
+pub fn generate(peripherals: &mut xmltree::Element, c: &chip::Chip) -> crate::Result<()> {
+    // Find the (first) peripheral with the name `CPU` to then add the interrupts to it.
+    for peripheral in peripherals.children.iter_mut() {
+        if peripheral.get_child("name".to_string()).and_then(|e| {
+            // Once `inner_deref` is stabilized, this can be changed to
+            // e.text.deref() or e.text.as_deref(), depending on the final name
+            e.text.as_ref().map(|s| s.as_ref())
+        }) == Some("CPU") {
+            peripheral.children.extend(c.interrupts.values().map(|interrupt| {
+                let mut el = xmltree::Element::new("interrupt");
+
+                el.children.push(xmltree::Element::new_with_text(
+                    "name",
+                    interrupt.name.clone(),
```

Comment:
You don't need to clone here, just use `&interrupt.name`.

Reply:
Unfortunately, this doesn't work, but `interrupt.name.as_ref()` does.